src/util/http/websocket/QueryToSocketDistributor.cpp

```diff
@@ -7,52 +7,78 @@
 #include <boost/asio/associated_cancellation_slot.hpp>
 #include <boost/asio/use_awaitable.hpp>
 
+#include "util/Log.h"
+
 namespace ad_utility::websocket {
 
 template <typename CompletionToken>
 auto QueryToSocketDistributor::waitForUpdate(CompletionToken&& token) {
-  auto initiate = [this]<typename Handler>(Handler&& self) mutable {
+  auto id = counter++;
```

Comment:
How is this threadsafe when the `counter` is not atomic?

Reply:
`waitForUpdate()` is private API and only ever called on the same strand